parsing/parse.mli

```diff
@@ -27,3 +27,68 @@ val use_file : Lexing.lexbuf -> Parsetree.toplevel_phrase list
 val core_type : Lexing.lexbuf -> Parsetree.core_type
 val expression : Lexing.lexbuf -> Parsetree.expression
 val pattern : Lexing.lexbuf -> Parsetree.pattern
+
+module Longident: sig
+  (** The functions below can be used to parse Longident safely.
+      The common optional [cmd] argument is used to set the location header
+      for parsing error messages if they are emitted.
+      All variants parse slightly different sets of {!Longident.t}.
+  *)
+
+  val value: ?cmd:string -> string -> (Longident.t,exn) result
```

Comment:
I'm not fond of the API.

I think that we should expose `Lexing.lexbuf -> Longident.t` functions, as above. In particular, Merlin does propagate location information up to the point where they want to parse longidents, so they would be able to initialize the lexbuf location in interesting ways.

We could provide a generic wrapper in `parse`, for example `(lexbuf -> 'a) -> ?cmd:string -> string -> ('a, exn) result`. I'm not convinced this is necessary (and it is doing many orthogonal things at once: taking string inputs, changing error messages, and reifying exceptions into the result monad).

Reply:
Indeed, `Lexing.lexbuf -> Longident.t` seems better, even for the compiler.